File: backend/amoz/src/main/java/com/zpi/amoz/requests/CustomerB2CCreateRequest.java
package com.zpi.amoz.requests;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

@Schema(description = "Request do tworzenia klienta B2C, zawierający dane klienta oraz dane osobowe klienta.")
public record CustomerB2CCreateRequest(

        @Schema(description = "Sekcja klienta, zawierająca dane klienta", implementation = CustomerCreateRequest.class)
        @NotNull(message = "Customer section must be filled")
        CustomerCreateRequest customer,

        @Schema(description = "Dane osobowe klienta", required = true, implementation = PersonCreateRequest.class)
        @NotNull(message = "Customer's personal data must be given")
        PersonCreateRequest person

) {
}

File: frontend/app/src/main/java/com/example/amoz/api/requests/CustomerB2CCreateRequest.kt
package com.example.amoz.api.requests

import com.example.validation.annotations.NotNullable
import com.example.amoz.interfaces.ValidatableRequest
import com.example.amoz.models.CustomerB2C
import kotlinx.serialization.Serializable

@Serializable
data class CustomerB2CCreateRequest(
    @field:NotNullable(nameOfField = "Customer")
    val customer: CustomerCreateRequest = CustomerCreateRequest(),

    @field:NotNullable(nameOfField = "Person")
    var person: PersonCreateRequest = PersonCreateRequest()
) : ValidatableRequest<CustomerB2CCreateRequest>() {
    constructor(customerB2C: CustomerB2C): this(
        customer = CustomerCreateRequest(customerB2C.customer),
        person = PersonCreateRequest(customerB2C.person)
    )
}
